azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -383,3 +390,174 @@ def _base64_encode(self, data):
         if PY_VERSION_MAJOR > 2:
             return s.decode('utf-8')
         return s
+
```

Comment:
Move from the WireClient, the only difference in the implementation is the use of ResourceGoneError to report 410s (added the exception here because the retry logic needs to be implemented by the caller)